tools/pipelines/templates/build-npm-package.yml

```diff
@@ -407,6 +412,41 @@ extends:
                         inputs:
                           sourceFolder: '${{ parameters.buildDirectory }}/nyc/report'
                           targetFolder: $(Build.ArtifactStagingDirectory)/codeCoverageAnalysis
+                      - task: Bash@3
+                        displayName: Report Code Coverage Comparison
+                        condition: and(succeededOrFailed(), eq('${{ parameters.reportCodeCoverageComparison }}', true), eq(variables['ReportDirExists'], 'true'), eq(variables['Build.Reason'], 'PullRequest'), eq(variables['System.PullRequest.TargetBranch'], 'main'))
+                        continueOnError: false
+                        env:
+                          ADO_API_TOKEN: $(System.AccessToken)
+                          GITHUB_API_TOKEN: $(githubPublicRepoSecret)
+                          TARGET_BRANCH_NAME: '$(targetBranchName)'
+                          ADO_BUILD_ID: $(Build.BuildId)
+                        inputs:
+                          targetType: inline
+                          workingDirectory: '${{ parameters.buildDirectory }}'
+                          script: |
+                            Coverage_Report_Artifact_Name="Code Coverage Report"
+                            export CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_BASELINE=$Coverage_Report_Artifact_Name
+                            export CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_PR=$Coverage_Report_Artifact_Name
+                            echo "Code Coverage Artifact Name Prefix for Baseline: $CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_BASELINE"
+                            echo "Code Coverage Artifact Name Prefix for PR: $CODE_COVERAGE_ANALYSIS_ARTIFACT_NAME_PR"
+                            REPO_NAME=$(echo $BUILD_REPOSITORY_NAME | cut -d'/' -f2)
+                            REPO_OWNER=$(echo $BUILD_REPOSITORY_NAME | cut -d'/' -f1)
+                            export GITHUB_REPOSITORY_NAME=$REPO_NAME
+                            export GITHUB_REPOSITORY_OWNER=$REPO_OWNER
+                            echo "Github Repository Owner: $GITHUB_REPOSITORY_OWNER"
+                            echo "Github Repository Name: $GITHUB_REPOSITORY_NAME"
+                            export GITHUB_PR_NUMBER=$SYSTEM_PULLREQUEST_PULLREQUESTNUMBER
+                            echo "Github PR number: $GITHUB_PR_NUMBER"
```

Comment:
Nit: pass this as `GITHUB_PR_NUMBER: $(System.PullRequest.PullRequestNumber)` from the `env:` property of the task? Since it doesn't require any special computation in the script.

I'd do the same for `ADO_CI_BUILD_DEFINITION_ID_BASELINE` and `ADO_CI_BUILD_DEFINITION_ID_PR` for the same reason. The less we have in the content of the script, the better, IMO.